### PR TITLE
Add handling of identical candidates

### LINKED ISSUE #2666: Uncaught exception in `/api/candidates`
```
ERROR:tornado.application:Uncaught exception POST /api/candidates (131.215.198.213)
HTTPServerRequest(protocol='http', host='fritz.science', method='POST', uri='/api/candidates', version='HTTP/1.0', remote_ip='131.215.198.213')
Traceback (most recent call last):
  File "/skyportal_env/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1802, in _execute_context
    self.dialect.do_execute(
  File "/skyportal_env/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 732, in do_execute
    cursor.execute(statement, parameters)
psycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "candidates_main_index"
DETAIL:  Key (obj_id, filter_id, passed_at)=(Gaia22and, 128, 2022-02-07 15:12:57.6) already exists.
```

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR handles the posting of identical candidates more gracefully.

Closes #2666.

### CODE REVIEW

File: skyportal/tests/api/test_candidates.py [hunk 1]
```diff
@@ -1504,3 +1504,48 @@ def test_candidate_list_pagination(
...
+            "redshift": 3,
+            "transient": False,
+            "ra_dis": 2.3,
+            "filter_ids": [public_filter.id],
+            "passed_at": passed_at,
+        },
+        token=upload_data_token,
+    )
+    # this time fails
+    assert status == 400
```
Comment (guynir42) [inline skyportal/tests/api/test_candidates.py:1551]: I would add a check that you get the correct error message (at least check that `Failed to post candidate` is in the message), just to make sure the error you think you are getting is the right one.

**Review by guynir42 — APPROVED** [review]
Nice work. A small addition below and then you can merge.

File: skyportal/tests/api/test_candidates.py [hunk 2]
```diff
@@ -1503,49 +1505,3 @@ def test_candidate_list_pagination(
         token=view_only_token,
     )
     assert status == 400
-    assert "Page number out of range" in data["message"]
```
Comment (guynir42) [inline skyportal/tests/api/test_candidates.py:1506]: Why is this line deleted...?

File: skyportal/tests/api/test_candidates.py [hunk 3]
```diff
@@ -216,7 +216,9 @@ def test_token_user_cannot_post_two_candidates_same_obj_filter_passed_at(
         },
         token=upload_data_token,
     )
-    assert status in [401, 500]
```
Comment (guynir42) [inline skyportal/tests/api/test_candidates.py:219]: Its quite amazing that we were actually testing for this and excepting the wrong return code. Good catch!

**Review by guynir42 — APPROVED** [review]

File: skyportal/tests/api/test_candidates.py (hunk 2 above)
Comment (mcoughlin) [reply skyportal/tests/api/test_candidates.py]: Because I don't know how copy and paste works.

**Comment by guynir42** [comment]
@mcoughlin merge when green.

**Comment by mcoughlin** [comment]
Thanks @guynir42

File: skyportal/tests/api/test_candidates.py (hunk 3 above)
Comment (stefanv) [reply skyportal/tests/api/test_candidates.py:219]: There that mystery is solved :)

**Comment by stefanv** [comment]
:rocket: Thanks, Michael!
